web/template.py

```diff
@@ -242,7 +242,7 @@ def read_keyword(self, text):
         line, text = splitline(text)
         return StatementNode(line.strip() + "\n"), text
 
-    def read_expr(self, text, escape=True):
+    def read_expr(self, text, escape=True):  # noqa: C901, PLR0915
```

Comment:
Can you please add the string that was causing you trouble into a new test in tests/test_template.py so we can ensure that we do not have any regressions?

Please drop the noqas in favor of increasing the corresponding values at
https://github.com/webpy/webpy/blob/ed3e92cceb6ed870b224107ea653f48fa7fc2d0a/pyproject.toml#L86-L93

Reply:
I did this, but I sure don't feel good about it! I also added commit to move some code out of `Parse` that would be easy to drop if undesired.